src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp

```diff
@@ -244,6 +244,10 @@ std::vector<std::string> disabledTestPatterns() {
         retVector.emplace_back(R"(.*Snippets.*MHA.*)");
         retVector.emplace_back(R"(.*Snippets.*(MatMul|Matmul).*)");
     }
+    if (!InferenceEngine::with_cpu_x86_avx512_core_fp16()) {
```

Comment:
Applied. Thanks Dmitry!

Reply:
@dmitry-gorokhov Hi Dmitry! When the above compilation condition is added, ci tests in [ie_tests_cpu_debian_9_arm64_release](https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/ie-tests-linux-debian_9_arm64-cpu/4922/testReport/junit/(root)/smoke_Reduce_OneAxis_CPU_ReduceCPULayerTest/CompareWithRefs_IS_______TS___2_19_2_9____axes__0__opType_SCALAR_type_Mean_KeepDims_true_netPRC_f32_inPRC_undefined_outPRC_undefined_PluginConf_INFERENCE_PRECISION_HINT_f16/) begin to fail, because ARM choose to execute fp32 instead of f16. I'm not sure, but I'm thinking that the condition [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/config.cpp#L182) also apply for ARM, and if the condition in  is not meet, fp32 will still be chosen rather than f16. Should we revert the compilation condition change, or maybe we also revise the condition [here](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/config.cpp#L182) to be       
`#if defined(OPENVINO_ARCH_X86) || defined(OPENVINO_ARCH_X86_64)`
`                if (mayiuse(avx512_core_fp16) || mayiuse(avx512_core_amx_fp16)) {`
`                    inferencePrecision = ov::element::f16;`
`                    inferencePrecisionSetExplicitly = true;`
`                }`
`#else`
`                inferencePrecision = ov::element::f16;`
`                inferencePrecisionSetExplicitly = true;`
`#endif`
What do you think? Thanks Dmitry!

Reply:
Hi Dmitry, I've applied the revision about this comment in a separate commit. Please feel free to comment. Thanks!

Reply:
Ok, In general to handle such cases we have test instances separated by CPU arch (x64/reduce, arm/reduce, common reduce). Since ARM doesn't support FP16 for now I would propose to add FP16 instances only for x64.

Reply:
@dmitry-gorokhov Hi Dmitry! I've reverted this revision. Because I find that this `InferenceEngine::with_cpu_x86_avx512_core_fp16()` has already been put under `#if defined(OPENVINO_ARCH_X86) || defined(OPENVINO_ARCH_X86_64)` in [system_conf.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/inference/src/system_conf.cpp#L37).  Just like [the counterpart of bf16](https://github.com/openvinotoolkit/openvino/blob/4d777bd65b125ae3ac94a8f06e35ced9b8fb51d0/src/plugins/intel_cpu/tests/functional/shared_tests_instances/skip_tests_config.cpp#L239), at a few lines above. By original design, it will return false directly for ARM, then the 'if' condition is true, so those cases will be skipped for ARM. It seems the same logic should be applied for fp16. Further discussions are always welcomed. Thanks Dmitry!

Reply:
> Ok, In general to handle such cases we have test instances separated by CPU arch (x64/reduce, arm/reduce, common reduce). Since ARM doesn't support FP16 for now I would propose to add FP16 instances only for x64.

@dmitry-gorokhov I missed this comment. I agree. Let me apply your proposal. Thanks Dmitry!

Reply:
We are adding FP16 on ARM in separate [PR](https://github.com/openvinotoolkit/openvino/pull/18394) and the logic you introduce here brakes it since you disable all FP16 tests for ARM. 
Again correct fix is to add FP16 instances only inside x64/reduce. 
We will never have aligned FP16 precision support between two CPU arch, so different test instances is preferable solution comparing to generic skip configuration.

Reply:
Applied! Thanks a lot Dmitry!